src/Platforms/PlatformSelector.hpp

```diff
@@ -0,0 +1,46 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2022 QMCPACK developers.
+//
+// File developed by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//
+// File created by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef QMCPLUSPLUS_PLATFORM_SELECTOR_H
+#define QMCPLUSPLUS_PLATFORM_SELECTOR_H
+
+#include <vector>
+#include <string>
+
+namespace qmcplusplus
+{
+
+enum class PlatformKind {
+  CPU,
+  OMPTARGET,
+  CUDA
+};
+
+enum class SelectorKind {
+  OMPTARGET,
+  OMPTARGET_AND_CUDA
```

Comment:
I probably should change these to "CPU_OMPTARGET" and "CPU_OMPTARGET_CUDA". I may also need "CPU_CUDA" and "OMPTARGET_CUDA".